.cursorrules

```diff
@@ -0,0 +1,530 @@
+- If asked to create a service, create ONLY the service file
+- Wait for explicit instructions before adding any additional functionality
+- Ask for clarification if the scope is unclear rather than assuming
+
+## Rule 6: Use Available Libraries
+
+CRITICAL REQUIREMENT:
+
+- Always attempt to use libraries that are listed in the `package.json` for relevant functionality
+- Avoid implementing custom solutions when a library can provide the needed functionality
+
+IMPLEMENTATION:
+
+- Check the `package.json` for available libraries before implementing new functionality
+- Use the library's documentation for the version `package-lock.json` specifies to integrate it effectively into the codebase
+
+## Rule 7: NestJS Services Directory Structure
+
+CRITICAL REQUIREMENT:
+
+- Every `*.service.ts` file MUST reside in a `services/` subdirectory within its module folder
+- This rule applies consistently across ALL modules in ANY NestJS application
+
+EXPECTED STRUCTURE:
+
+```
+src/module-name/
+  module-name.module.ts
+  module-name.controller.ts
+  services/
+    module-name.service.ts
+    other-service.service.ts
+```
+
+PROHIBITED:
+
+- Placing service files directly in module root
+- Creating service files outside a `services/` folder
+- Having multiple `services/` folders per module
+- Any exceptions or deviations from this pattern
+
+## Rule 8: Use http-constants-ts for MIME Types
+
+CRITICAL REQUIREMENT:
+
+- Always use the `http-constants-ts` library for MIME type constants
+- Avoid using string literals for MIME types in the codebase
+
+IMPLEMENTATION:
+
+- Import the necessary MIME type constants from `http-constants-ts`
+- Replace any existing string literals representing MIME types with the corresponding constants from the library
+
+## Rule 9: Consume Provided Documentation Links
+
+CRITICAL REQUIREMENT:
+
+- Always read and utilize documentation links provided by the user
+- Ensure that the information from the links is accurately reflected in the code changes or responses
+
+IMPLEMENTATION:
+
+- Thoroughly review the content of any documentation links provided by the user
+- Apply the information from the documentation to the task at hand, ensuring accuracy and relevance
+
+## Rule 10: Mandatory Rule Compliance
+
+CRITICAL REQUIREMENT:
+ALL rules, both global and project-specific, MUST be followed with ZERO exceptions.
+
+REQUIRED BEHAVIOR:
+
+```javascript
+// Rule compliance is MANDATORY
+const beforeAnyAction = () => {
+  checkAllGlobalRules()
+  checkAllProjectRules()
+  verifyExplicitPermissions()
+  if (!allRulesFollowed || !explicitPermission) {
+    HALT_IMMEDIATELY()
+  }
+}
+```
+
+VERIFICATION CHECKLIST:
+
+1. ✓ Verify compliance with ALL global rules
+2. ✓ Verify compliance with ALL project rules
+3. ✓ Confirm EXPLICIT permission exists
+4. ✓ Cross-reference against ALL rules
+5. ✓ If ANY uncertainty exists, STOP
+
+STRICTLY PROHIBITED:
+
+- Creating ANY files without EXPLICIT permission
```

Comment:
I rather have the agent create the file and me reviewing them instead of ask permission for each.